### PR TITLE
fix: improve all-day to timed event switch experience

### LINKED ISSUE #917: Default time spann on disabling all-day event flag
### Checklist

- [x] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Calendar/issues) or [closed](https://github.com/FossifyOrg/Calendar/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [x] I made sure that there are **no existing discussions** - [open](https://github.com/orgs/FossifyOrg/discussions?discussions_q=is:open+label:calendar) or [closed](https://github.com/orgs/FossifyOrg/discussions?discussions_q=is:closed+label:calendar) - to which I could contribute my information.
- [x] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [x] **I have taken the time to fill in all the required details. I understand that the request will be dismissed otherwise.**
- [x] This issue contains only one feature request.
- [x] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Calendar/blob/master/CONTRIBUTING.md).

### Feature description

Event should consider the default event duration on disabling the all-day flag of an event.

### Why do you want this feature?

On disabling the all-day switch, an event which was previously all-day is timed from 0:00 to 12:00. When you change the start time to 11:00 the event will last to 23:00 preserving the 12-hour length of the event. 
However, when choosing a start time after noon, this will cause the event to span into the next day, which therefore requires too many clicks to select the correct end date and time. 
It would be more appropriate to use the default event duration used also on creating new events.

### Additional information

It needs to be decided whether the time span should be from `0:00` till `0:00 + defaultDuration` or `defaultStartTime` till `defaultStartTime + defaultDuration`. The second would probably be more consistent.

### PR DESCRIPTION
**Pull request by fm-sys**
#### Type of change(s)
- [x] Bug fix
- [ ] Feature / enhancement
- [ ] Infrastructure / tooling (CI, build, deps, tests)
- [ ] Documentation

#### What changed and why

If editing an all-day event, set default times so when user disables all-day,
the event uses proper default start time and duration while making sure to preserve the correct end date

#### Tests performed

 - tested several edge cases including multi-day all-day events

#### Closes the following issue(s)

- Closes #917

#### Checklist
- [x] I read the [contribution guidelines](../blob/HEAD/CONTRIBUTING.md).
- [x] I manually tested my changes on device/emulator (if applicable).
- [x] I updated the "Unreleased" section in `CHANGELOG.md` (if applicable).
- [x] I have self-reviewed my pull request (no typos, formatting errors, etc.).
- [x] All checks are passing.

### CODE REVIEW

File: app/src/main/kotlin/org/fossify/calendar/activities/EventActivity.kt [hunk 1]
```diff
@@ -713,6 +714,27 @@ class EventActivity : SimpleActivity() {
...
+                0
+            )
+            mEventEndDateTime = mEventEndDateTime.withTime(
+                endTime.hourOfDay,
+                endTime.minuteOfHour,
+                0,
+                0
+            )
+        }
+
```
Comment (naveensingh) [inline app/src/main/kotlin/org/fossify/calendar/activities/EventActivity.kt]: It'll be safer to do this one-time migration directly in `toggledAllDay()`. Right now, this block will run unconditionally and might lead to subtle bugs later (or now if existing code assumes `mEventStartDateTime` and `mEventEndDateTime` reflect the DB values).

**Comment by fm-sys** [comment]
> The new event end time should be 2 AM, 4 Dec, but it's 8 PM, 3 Dec

Setting it to _2 AM, 4 Dec_ would have a mayor downside: Disabling and re-enabling all-day would convert a single day event into a two day event which would IMHO be significantly worse than ignoring the default duration.

Sure we could detect this case and add some special handling, but wouldn't this make things even more inconsistent?

**Comment by naveensingh** [comment]
> Sure we could detect this case and add some special handling, but wouldn't this make things even more inconsistent?

I think it would be more consistent if handled properly, but I agree that adding special handling for this rare edge case would complicate this switch more than necessary. For now, let's stick with 8 PM, 3 Dec, and we can revisit if it is ever reported.

**Comment by naveensingh** [comment]
BTW, this should also be fixed for tasks.

**Comment by fm-sys** [comment]
For the forks main branch, I already stick with merge. 

For providing individual PRs to the Fossify repo, I indeed rebased the feature branches. In fact, I contributed to open source projects before who preferred rebased branches which can be merged via fast-forward. But for sure, I can stick to merge for this repo! Thank you so much for merging by PRs btw :)
